### ISSUE TITLE
why there is logits_per_image.t() in contrastive loss?

### ISSUE THREAD

**Issue by hyeonjinXZ** [adymaharana/VLCStoryGan#10]
Hello, there!

In the https://github.com/adymaharana/VLCStoryGan/blob/74112404689e8144c2ed2d375e1e5a1cde09debb/vlcgan/miscc/utils.py#L163, there is two kinds of loss values (loss_i and loss_t) in line 184 and 185.
And the function calculate the average of them.
https://github.com/adymaharana/VLCStoryGan/blob/74112404689e8144c2ed2d375e1e5a1cde09debb/vlcgan/miscc/utils.py#L186

**Why to calculate like this? I didn't figure out the meaning of this. And what is the difference with the loss below?**
`loss = loss_fct(logits_per_image, labels)`

**Comment by hyeonjinXZ** [comment]
one is for image, and the other is for text.
the reference is the figure 3 in the paper 'Learning Transferable Visual Models From Natural Language Supervision'.
